src/msgpack_asgi/_middleware.py

```diff
@@ -1,25 +1,44 @@
 import json
+from typing import Any, Callable
 
 import msgpack
 from starlette.datastructures import Headers, MutableHeaders
 from starlette.types import ASGIApp, Message, Receive, Scope, Send
 
 
 class MessagePackMiddleware:
-    def __init__(self, app: ASGIApp) -> None:
+    def __init__(
+        self,
+        app: ASGIApp,
+        *,
+        packb: Callable[[Any], bytes] = msgpack.packb,
```

Comment:
Ah, so. Previously we were using `msgpack.packb(..., raw=False)`. It's interesting that we were able to change that to drop `raw=False`, and still have tests pass. Makes me doubt whether a/ that was necessary or b/ we're missing a test that asserts the necessity of that flag in certain cases.

IIRC, `raw=False` was meant so that `unpackb()` returns `str`, rather than raw `bytes`.

I see it got turned on by default in `msgpack-python` 1.0.0: https://github.com/msgpack/msgpack-python#major-breaking-changes-in-msgpack-10

> NOTE: In examples below, I use raw=False and use_bin_type=True for users using msgpack < 1.0. These options are default from msgpack 1.0 so you can omit them.

So, on `msgpack>=1` we get:

```python
>>> import msgpack
>>> msgpack.unpackb(msgpack.packb("hello"))
"hello"
```

But on `msgpack<1` we were getting:

```python
>>> import msgpack
>>> msgpack.unpackb(msgpack.packb("hello"))
b"hello"
>>> msgpack.unpackb(msgpack.packb("hello"), raw=False)
"hello"
```

And the reason our tests pass is because `msgpack` isn't pinned in `requirements.txt`, so it uses the latest, which is `>=1` now, which doesn't care about `raw=False` because it sets it by default. If I force-install `pip install msgpack<1`, I do get an error when running the tests bc the `json` is getting bytes:

```python
E       TypeError: keys must be str, int, float, bool or None, not bytes
```

So, I _think_, for the sake of risk-aversion, we should keep the previous behavior (using `raw=False`, which happens to be the default for `msgpack>=1`) in this PR, and treat possibly dropping that in a separate issue. Yes?

If so, we should use:

```python
from functools import partial
```

```suggestion
        packb: Callable[[Any], bytes] = partial(msgpack.packb, raw=False),
```

Reply:
^Woops, I meant to target the `unpackb` parameter here, not `packb`.

Reply:
Sure I agree :rocket: ! 
functools's partial wasn't imported tho. Can you add another suggestion?

Reply:
> ^Woops, I meant to target the unpackb parameter here, not packb.
Added it in the last commit!